virttest/utils_misc.py

```
@@ -4698,3 +4698,77 @@ def _md5(fd):
+            with open(cvm_module_path) as f:
+                output = f.read().strip()
+                if output not in params.objects("module_status"):
+                    raise exceptions.TestCancel(
+                        f"Host support for {cvm_type} capability check failed."
+                    )
+        else:
+            raise exceptions.TestCancel(
+                f"Host support for {cvm_type} capability check failed."
+            )
+    except IOError as e:
+        raise exceptions.TestCancel(f"Failed to read {cvm_module_path}: {str(e)}")
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)